### PR TITLE
Update parameter prefix for query filter

### LINKED ISSUE #9850: Query: Parameter Extraction can cause collision in parameter name causing crash
Repro code:
```C#
using System;
using System.Linq;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace EFSampleApp
{
    public class Program
    {
        public static void Main(string[] args)
        {
            using (var db = new MyContext())
            {
                // Recreate database
                db.Database.EnsureDeleted();
                db.Database.EnsureCreated();

                // Seed database


                db.SaveChanges();
            }

            using (var db = new MyContext())
            {
                // Run queries
                var Enabled = true;
                var query = db.Blogs.Where(b => b.IsDeleted == Enabled).ToList();

            }

            Console.WriteLine("Program finished.");
        }
    }


    public class MyContext : DbContext
    {
        // Declare DBSets
        public DbSet<Blog> Blogs { get; set; }

        public bool Enabled { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            // Select 1 provider
            optionsBuilder
                .UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=_ModelApp;Trusted_Connection=True;Connect Timeout=5;ConnectRetryCount=0")
                //.UseSqlite("filename=_modelApp.db")
                //.UseInMemoryDatabase(databaseName: "_modelApp")
                .EnableSensitiveDataLogging()
                .UseLoggerFactory(new LoggerFactory().AddConsole(LogLevel.Trace));
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            // Configure model
            modelBuilder.Entity<Blog>().HasQueryFilter(e => e.IsDeleted == Enabled);
        }
    }

    public class Blog
    {
        public int Id { get; set; }
        public bool IsDeleted { get; set; }
    }
}

```

Exception/Stack Trace
```
dbug: Microsoft.EntityFrameworkCore.Query[100101]
      Compiling query model:
      'from Blog b in DbSet<Blog>
      where [b].IsDeleted == __Enabled_0
      select [b]'
dbug: Microsoft.EntityFrameworkCore.Query[100104]
      Optimized query model:
      'from Blog b in DbSet<Blog>
      where [b].IsDeleted == __Enabled_0
      where [b].IsDeleted == __Enabled_0
      select [b]'
dbug: Microsoft.EntityFrameworkCore.Query[100107]
      (QueryContext queryContext) =>
      {
          context = (MyContext)queryContext.Context
          Void queryContext.AddParameter(
              name: "__Enabled_0",
              value: (object)Invoke(context => context.Enabled, context)
          )
          return IEnumerable<Blog> _InterceptExceptions(
              source: IEnumerable<Blog> _TrackEntities(
                  results: IEnumerable<Blog> _ShapedQuery(
                      queryContext: queryContext,
                      shaperCommandContext: SelectExpression:
                          SELECT [b].[Id], [b].[IsDeleted]
                          FROM [Blogs] AS [b]
                          WHERE ([b].[IsDeleted] = @__Enabled_0) AND ([b].[IsDeleted] = @__Enabled_0),
                      shaper: UnbufferedEntityShaper<Blog>),
                  queryContext: queryContext,
                  entityTrackingInfos: { itemType: Blog },
                  entityAccessors:
 List<Func<Blog, object>>
                  {
                      Func<Blog, Blog>,
                  }
              ),
              contextType: EFSampleApp.MyContext,
              logger: DiagnosticsLogger<Query>,
Unhandled Exception:              queryCon text: queryContext)
      }

System.ArgumentException: An item with the same key has already been added. Key: __Enabled_0
   at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(Object key)
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.EntityFrameworkCore.Query.QueryContext.AddParameter(String name, Object value)
   at lambda_method(Closure , QueryContext )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
   at Remotion.Linq.QueryableBase`1.GetEnumerator()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at EFSampleApp.Program.Main(String[] args) in D:\code\EFSampleApp\EFSampleApp\Program.cs:line 28
```

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #9850

### CODE REVIEW

File: src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs [hunk 1]
```diff
@@ -437,8 +437,8 @@ public virtual object Evaluate([CanBeNull] Expression expression, out string par
                 if (newExpression != expression)
                 {
                     parameterName = newExpression is MemberExpression memberExpression
-                        ? memberExpression.Member.Name
-                        : "_queryFilter";
+                        ? "$" + memberExpression.Member.Name
+                        : "$";
```
Comment (anpete) [inline src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: DRY "$"
How about "filter" as base name for non-member case?
Do we have test for non-member case?

**Comment by smitpatel** [comment]
@anpete - Updated.

**Comment by smitpatel** [comment]
@anpete - ping

File: src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs [hunk 2]
```diff
@@ -436,9 +436,9 @@ public virtual object Evaluate([CanBeNull] Expression expression, out string par
 
                 if (newExpression != expression)
                 {
-                    parameterName = newExpression is MemberExpression memberExpression
-                        ? memberExpression.Member.Name
-                        : "_queryFilter";
+                    parameterName = "$" + (expression is MemberExpression memberExpression
+                                        ? memberExpression.Member.Name
+                                        : "Filter");
```
Comment (anpete) [inline src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: NIT: "filter"

**Pull request merged**